src/index.ts

```diff
@@ -143,25 +150,25 @@ export class AssembleSubmitButtonExtension
     });
     panel.toolbar.addItem('githubAuthButton', githubAuthButton);
 
-    // ugh
-    // TODO - remove callback when authenticated, assuming that we will never become un-authenticated
-    let authIntervalId = setInterval(authCallback, 5 * ONE_SECOND);
-
-    async function authCallback() {
-      let authenticated = await isAuthenticated();
+    async function authCallback(popup?: Window, authIntervalId?: number) {
+      const authenticated = await isAuthenticated();
       githubAuthButton.toggleClass(
         'assemble-githubAuthButtonIcon-authenticated',
         authenticated
       );
       if (authenticated) {
         // githubAuthButton.update = 'Already authenticated with GitHub';
-        clearInterval(authIntervalId);
-        if (!popup.closed) {
+        if (authIntervalId) {
+          clearInterval(authIntervalId);
+        }
+        if (popup && !popup.closed) {
           popup.close();
         }
       }
     }
 
+    authCallback().catch(console.warn);
```

Comment:
add a comment indicating the logic behind this line

Reply:
are you referring to the authCallback function being called there in general, or `.catch(console.warn)`? the latter was only added to satisfy the `no-floating-promises` rule

Reply:
referring to the authCallback invocation, since it is a "callback" but invoked here directly. You can give the reader a brief explanation

Reply:
maybe the name callback doesn't fit the function well anymore with my changes. but the best I can come up with would be calling it "authHandler". It does too many things at this point to give it a good name. I've thought about splitting it up into multiple functions (as we only need to clear the interval and close the popup when called by the interval, otherwise it's just the button class) but found no elegant way of doing so without adding unnecessary cognitive complexity to the code.

Reply:
Fair enough, the comment is helpful as an intermediate solution